### PR TITLE
GH-24: Enable > 1 Partitions in @TopicPartition

### LINKED ISSUE #24: @TopicPartition:partition -> partitions
Allow a single `@TopicPartition` annotation to support > 1 partition on the same topic.

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves #24

### CODE REVIEW

**Comment by artembilan** [comment]
Travis + Checkstyle is our "foe" :smile:

**Comment by garyrussell** [comment]
I say "friend" not "foe".; pushed

File: spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java [hunk 1]
```diff
@@ -434,6 +438,35 @@ else if (resolvedValue instanceof Iterable) {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
+	private void resolvePartitionAsInteger(String topic, Object resolvedValue, List<org.apache.kafka.common.TopicPartition> result) {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java]: This line of code exceeds 120 length.
Will fix on merge.

File: spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java [hunk 2]
```diff
@@ -216,7 +230,7 @@ public void listen2(@Payload String foo, @Header(KafkaHeaders.PARTITION_ID) int
 			this.latch2.countDown();
 		}
 
-		@KafkaListener(id = "baz", topicPartitions = @TopicPartition(topic = "annotated3", partition = "0"))
+		@KafkaListener(id = "baz", topicPartitions = @TopicPartition(topic = "annotated3", partitions = "0"))
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java]: Maybe `topic` in the `@TopicPatition` has to be required, not `optional` with `default ""`?
```
10:49:02.070 TRACE [bar-0-kafka-1][org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer] Polling...
10:49:02.113 WARN  [baz-0-kafka-1][org.apache.kafka.clients.NetworkClient] Error while fetching metadata with correlation id 138 : {=INVALID_TOPIC_EXCEPTION}
```

Comment (garyrussell) [inline spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java]: Yeah - also a bug in the test - waiting for latch 4 not 5; will fix - I want to make a couple more polishes too...

**Comment by garyrussell** [comment]
Pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/a6a71a44f54ba65233e095db786048721cd66248

**Pull request closed without merging**